examples/platform/silabs/wifi/icd/ApplicationSleepManager.cpp

```diff
@@ -92,24 +102,93 @@ bool ApplicationSleepManager::CanGoToLIBasedSleep()
         ChipLogProgress(AppServer, "Commissioning Window is Open - Cannot go to LI based sleep");
         canGoToLIBasedSleep = false;
     }
+    else if (mIsInActiveMode)
+    {
+        ChipLogProgress(AppServer, "Device is in active mode - Cannot go to LI based sleep");
+        canGoToLIBasedSleep = false;
+    }
     else
     {
         for (auto it = mFabricTable->begin(); it != mFabricTable->end(); ++it)
         {
             if (!mSubscriptionsInfoProvider->FabricHasAtLeastOneActiveSubscription(it->GetFabricIndex()))
             {
-                ChipLogProgress(AppServer, "Fabric index %u has no active subscriptions - Cannot go to LI based sleep",
-                                it->GetFabricIndex());
-                canGoToLIBasedSleep = false;
-                break;
+                ChipLogProgress(AppServer, "Fabric index %u has no active subscriptions", it->GetFabricIndex());
+                canGoToLIBasedSleep = ProcessSpecialVendorIDCase(it->GetVendorId());
+
+                if (canGoToLIBasedSleep)
+                {
+                    ChipLogProgress(AppServer,
+                                    "Fabric index %u with vendor ID : %d  has an edge case that allows for LI based sleep",
+                                    it->GetFabricIndex(), it->GetVendorId());
+                }
+                else
+                {
+                    // Fabric doesn't meet the requirements to allow us to go to LI based sleep
+                    break;
+                }
             }
-            ChipLogProgress(AppServer, "Fabric index %u has an active subscriptions!", it->GetFabricIndex());
         }
     }
 
     return canGoToLIBasedSleep;
 }
 
+bool ApplicationSleepManager::ProcessSpecialVendorIDCase(chip::VendorId vendorId)
+{
+    bool hasValidException = false;
+    switch (to_underlying(vendorId))
+    {
+    case to_underlying(SpecialCaseVendorID::kAppleKeychain):
+        hasValidException = ProcessKeychainEdgeCase();
+        break;
+
+    default:
+        break;
+    }
+
+    return hasValidException;
+}
+
+bool ApplicationSleepManager::ProcessKeychainEdgeCase()
+{
+    bool hasValidException = false;
+
+    for (auto it = mFabricTable->begin(); it != mFabricTable->end(); ++it)
+    {
+        if (to_underlying(it->GetVendorId()) == chip::VendorId::Apple &&
```

Comment:
Are you sure you need to_underlying? I think GetVendorId() returns chip::VendorId.
```suggestion
        if (it->GetVendorId() == chip::VendorId::Apple &&
```